### PR TITLE
Filter None from report, catch all possible exceptions during upload and do not stage the project

### PR DESCRIPTION
**Pull request by aanil**

### CODE REVIEW

File: daily_read/__main__.py [hunk 1]
```diff
@@ -93,20 +93,20 @@ def generate_all(upload=False, develop=False):
...
-                    if uploaded:
-                        op.projects_data.stage_data_for_project(project)
+            # Publish reports with published, hide reports with review
+            for upload_category, report_state in {"projects": "published", "delete_report_for": "review"}.items():
+                for status in modified_orders[owner][upload_category].keys():
+                    for project in modified_orders[owner][upload_category][status]:
+                        uploaded = False
+                        report_upload = report if upload_category == "projects" else ""
+                        try:
+                            uploaded = op.upload_report_to_order_portal(report_upload, project, report_state)
```
Comment (alneberg) [inline daily_read/__main__.py:103]: I think the log message inside this function will still say that the report was updated even if the report is empty string, so perhaps we could update that log message to be more clear when it's the emtpy string?

File: daily_read/__main__.py [hunk 2]
```diff
@@ -93,20 +93,20 @@ def generate_all(upload=False, develop=False):
...
+                            # Stage changes only if report was uploaded
+                            if uploaded:
+                                op.projects_data.stage_data_for_project(project)
+                        # catch any and every exception during upload
+                        except Exception as e:
+                            log.error(f"Theres was an issue in uploading/hiding reports for {project.project_id}: {e}")
```
Comment (alneberg) [inline daily_read/__main__.py]: Maybe make this error message more threatening. "Exception raised" and "Still continuing with other reports" and similar phrases could be used.

File: daily_read/__main__.py (hunk 1 above)
Comment (aanil) [reply daily_read/__main__.py:103]: Updated the message in `order_portal.py`

File: daily_read/__main__.py (hunk 2 above)
Comment (aanil) [reply daily_read/__main__.py]: Made it a bit more threatening 😄

File: daily_read/order_portal.py [hunk 3]
```diff
@@ -162,7 +162,8 @@ def upload_report_to_order_portal(self, report, project, status):
         response = requests.post(url, headers=self.headers, json=indata)
 
         if response.status_code == 200:
-            log.info(f"Updated report for order with project id: {project.project_id}")
+            operation = "Updated" if report else "Hid"
```
Comment (alneberg) [inline daily_read/order_portal.py:165]:
```suggestion
            operation = "Updated" if report else "Hide"
```

**Review by alneberg — APPROVED** [review]
Thank you!

**Pull request merged**